### DISCUSSION TITLE
Allow the user to insert custom metadata?

### DISCUSSION THREAD

**Discussion by augustebaum** [probabl-ai/skore#574] [Ideas]
Something like
```python

fig, ax = plt.subplots()
...

Store("root").insert("my_plot", fig, custom_metadata={"description": "this is a plot showing things"})
```

**Comment by rouk1** [comment]
I would go for a simple `metadata` keyword argument. In the serialized version it may be named `user_metadata`.

**Comment by augustebaum** [comment]
Yes I didn't use `metadata` because we already use this name in `Store.read(key, metadata=True)` to retrieve a value and all its metadata (i.e. the default metadata like `created_at`).

**Comment by tuscland** [comment]
I'm not sure if I like metadata=True. Why not always return the metadata as a second return value?

**Comment by rouk1** [comment]
I agree with @tuscland I'm not fond of `metadata=True` it should be at least `include_metadata=True` or a second return value.

**Comment by augustebaum** [comment]
> Why not always return the metadata as a second return value?

You mean `value, metadata = Mandr.read("key")`? I don't think metadata is that important

**Comment by tuscland** [comment]
That's why :), by default you opt not to get it: `value = Mandr.read("key")`

**Comment by tuscland** [comment]
I think it is useful to let users add their own metadata.

This discussion relates to #120.

Also it requires UI/UX work to access metadata and filter items given metadata values.

**Comment by tuscland** [comment]
A possible workaround right now is to pair an item with an altername key suffixed with e.g. `_metadata` and store a primitive dictionary.

For example:
```python
project.put("myplot", plot)
project.put("myplot_metadata", { "notes": "It was fun to generate this plot." })
```

**Comment by MarieSacksick** [comment]
We started to implement notes. We might want to add tags at some point, maybe through the concept of a run.   
I feel that this discussion can be closed now.
